### PR TITLE
Added 12 'Must-Have' User stories

### PR DESCRIPTION
**Pull request by edwinghy**

### CODE REVIEW

**Comment by chuaweiwen** [comment]
It looks alright, based on what we have discussed in the earlier meeting. I notice some repeated stories on 'Nice-to-have' and 'Unlikely-to-have' ones

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 [width="59%",cols="22%,<23%,<25%,<30%",options="header",]
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
+|`* * *` |busy user |search using accumulative filters |narrow down my search in the shortest time possible
+
+|`* * *` |careless user that cant remember the order of entry |add a contact without following the particular order of entry |not remember the order of entry
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: "careless user that find it troublesome to follow a particular order of entering contact information"
"add contact information without following any particular order"
"enter contact information more easily and conveniently"

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |forgetful user that cant remember all tags |view a list of existing tags |recall the tags associated with a person
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: "can't"
I think the benefit is that we can recall and reuse tags we have used? So we won't have duplicate tags that are not exactly the same, but mean the same thing, like "cs2103" and "CS2103"

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |organized user |search for contacts by a given tag |get the list of contacts with the same tag
+
+|`* * *` |sociable user that was given contacts by another person |import from another source file |add contacts that were shared with me
+
+|`* * *` |sociable user |export selected contacts |share the contacts with friends
+
+|`* * *` |sociable user |get reminders of upcoming birthdays |plan my schedule
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: Something similar was done by Weiwen. Do you guys mean the same thing?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |careless user that cant remember the order of entry |add a contact without following the particular order of entry |not remember the order of entry
+
+|`* * *` |forgetful user that cant remember all tags |view a list of existing tags |recall the tags associated with a person
+
+|`* * *` |organized user |search for contacts by a given tag |get the list of contacts with the same tag
+
+|`* * *` |sociable user that was given contacts by another person |import from another source file |add contacts that were shared with me
+
+|`* * *` |sociable user |export selected contacts |share the contacts with friends
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: "share my contacts with friends"

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |sociable user |get reminders of upcoming birthdays |plan my schedule
+
+|`* * *` |user |sort the list by name alphabetically |
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: This one is also covered by Weiwen

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |user with limited details of a contact |add user with just the name and phone number |store acquaintance with limited details
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: "store contact information with limited detail"
I think we try to be consistent.

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -721,6 +721,30 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* * *` |user with large number of tags |identify tags by colors and their alphabetical order |easily identify different tags
```
Comment (justinpoh) [inline docs/DeveloperGuide.adoc]: "....and their alphabetical order in the GUI"

File: docs/DeveloperGuide.adoc (hunk 5 above)
Comment (chuaweiwen) [reply docs/DeveloperGuide.adoc]: There is another sort by name user stories by default at Line 758. We need to remove it.

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (chuaweiwen) [reply docs/DeveloperGuide.adoc]: There is a contradiction to this line. We have a Nice-to-have feature that is to store birthday to a contact, yet the birthday reminder is of higher importance.

I think we can assign "store birthday" as a must-have, while the reminders as "Nice-to-have"

File: docs/DeveloperGuide.adoc (hunk 4 above)
Comment (tshradheya) [reply docs/DeveloperGuide.adoc]: Club Import and Export into single one calling it "share contacts details"

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (chuaweiwen) [reply docs/DeveloperGuide.adoc]: Change this to sociable user |store the person's birthday |I can keep track of the person's birthday

**Review by chuaweiwen — CHANGES_REQUESTED** [review]

**Pull request merged**